README.md

```diff
@@ -246,6 +246,56 @@ Week^ {#353
 }
 ```
 
+### Collection Classes
+
+Similar to arrays, you might want to use collection classes. You can do so as long as your classes implement the `Traversable` interface.
+In this case, pjson will by default try to construct your class by passing in the array of data in the constructor.
+If this doesn't work for you, you can specify a custom factory method for your collections:
+
+```php
+class Collector
+{
+    use JsonSerialize;
+
+    #[Json(type: Schedule::class)]
+    public Collection $schedules;
+
+    #[Json(type: Schedule::class, collection_factory_method: '::make')]
```

Comment:
Not sure I like the `::` prefix. Can it just accept a callable?

```php
#[Json(type: Schedule::class, collection_factory: Collection::make(...))]
```

Reply:
Doesn't look like this works.
First off the syntax is only in 8.1+ and not in 8.0.
Second it doesn't seem to be valid syntax when used in an attribute.

Reply:
Ahh, that's a bummer.

Reply:
Though, I think supporting alternate parameters for static vs instance methods might be better than the special `::methodName` notation.

Reply:
Not sure about that. That would mean you could set both and then what ... You can throw an error, but I'd rather define an API that doesn't drive you to an error as much as possible.

Reply:
Can you dig in a bit to what's causing you pause with the `::` ?

Reply:
Can you instead check if the method is static vs non-static via reflection?

Reply:
I can indeed